src/appkit.cs

```diff
@@ -5324,8 +5506,20 @@ partial interface NSDocument {
 		[Export ("fileNameExtensionForType:saveOperation:")]
 		string FileNameExtensionForSaveOperation (string typeName, NSSaveOperationType saveOperation);
 
+		// Found in NSUserInterfaceValidations protocol with INSValidatedUserInterfaceItem param but bound originally with NSObject
+		// Adding protocol gave warning 0108 which is unfixable without API break 
+#if XAMCORE_4_0
 		[Export ("validateUserInterfaceItem:")]
-		bool ValidateUserInterfaceItem (NSObject /* Must implement NSValidatedUserInterfaceItem */ anItem);
+		bool ValidateUserInterfaceItem (INSValidatedUserInterfaceItem anItem);
+#else
+		[Export ("validateUserInterfaceItem:")]
+		bool ValidateUserInterfaceItem (NSObject /* must implement NSValidatedUserInterfaceItem */ anItem);
+
+#pragma warning disable 0108
+		[Wrap ("ValidateUserInterfaceItem ((NSObject)anItem)")]
+		bool ValidateUserInterfaceItem (INSValidatedUserInterfaceItem anItem);
```

Comment:
hmm... I wonder if it's not easier/safer to add `ValidateUserInterfaceItemn(IntPtr)` and implement both versions (`NSObject` and `INSValidatedUserInterfaceItem`) on top of it. `IsVirtual=true` can be used to maintain binary compatibility

Reply:
So I believe ValidateUserInterfaceItem is called by Cocoa to validate that certain UI items are appropriate in your current state. The docs say "Revert and Save" for now.

We could expose and ValidateUserInterfaceItemn (IntPtr) and handle it ourselves, but even if we renamed the ValidateUserInterfaceItem (INSValidatedUserInterfaceItem) to something that could live side by side with the NSObject one, would we call both from our IntPtr trampoline? What if they both returned different answers?

I'm happy to rework this, I just don't follow the trick.

Reply:
Yeah looks like this is called by Cocoa :( doing IntPtr here might help just a little doing `Runtime.GetINativeObject<T>` tho but not sure if it is really worth it